CorsixTH/Src/persist_lua.cpp

```diff
@@ -613,10 +609,10 @@ class lua_persist_basic_writer : public lua_persist_writer {
 
  private:
   lua_State* L;
-  uint64_t next_index;
+  uint64_t next_index{1};
   std::string data;
-  size_t data_size;
-  bool had_error;
+  size_t data_size{0};
+  bool had_error{false};
```

Comment:
Not sure I like this initialization. It has magic values seemingly copied from `init()`. Are these supposed to stay in sync with each other?

`had_error` is probably fine, as you cannot go back to "didn't have an error". But then, the initialization should only be here, right?

Reply:
It is not easy, perhaps not possible, to eliminate the init() method in persist_lua because of it's interaction with the lua state which involves referencing the userdata for the class.

However I have eliminated the redundant initialization of member variables pointed out above.

Reply:
I don't remember, but `init()` could also be used to draw a fresh graph while the game is running

Reply:
Maybe it should be `reset` then.